frontend/src/metabase/sharing/components/PulsesListSidebar.jsx

```diff
@@ -79,54 +91,103 @@ function PulsesListSidebar({
   );
 }
 
-PulsesListSidebar.propTypes = {
+_PulsesListSidebar.propTypes = {
   pulses: PropTypes.array.isRequired,
   createSubscription: PropTypes.func.isRequired,
   onCancel: PropTypes.func.isRequired,
   editPulse: PropTypes.func.isRequired,
+  parameters: PropTypes.array.isRequired,
 };
 
-function EmailRecipients({ pulse }) {
-  const recipients = pulse.channels[0].recipients;
-  const [first, ...rest] = recipients;
+function buildRecipientText(pulse) {
+  const {
+    channels: [firstChannel],
+  } = pulse;
+  if (firstChannel.channel_type !== "email") {
+    return "";
+  }
+
+  const [firstRecipient, ...otherRecipients] = firstChannel.recipients;
+  const firstRecipientText = firstRecipient.common_name || firstRecipient.email;
+  return _.isEmpty(otherRecipients)
+    ? firstRecipientText
+    : `${firstRecipientText} ${ngettext(
+        msgid`and ${otherRecipients.length} other`,
+        `and ${otherRecipients.length} others`,
+        otherRecipients.length,
+      )}`;
+}
 
-  let text = "";
+function buildFilterText(pulse, parameters) {
```

Comment:
This is EE-only code too, right? It's not super long, but probably best to move it too.

Reply:
This still gets used by OSS pulses since default parameters still apply to them

Reply:
👍 Ah ok cool.